server/web.js

```diff
@@ -314,3 +381,427 @@ function getPluginVersion(callback) {
+
+    // Add the last diff
+    var chunkedDiff = diff.diffLines(
+      currentBaseline.data.text.trim(), currentDoc.data.text.trim());
+
+    // Only push diffs with changes
+    if (!(chunkedDiff.length == 1 &&
+        !chunkedDiff[0].added &&
+        !chunkedDiff[0].removed)) {
+      chunkedDiffs.push(chunkedDiff);
+    }
+
+    return chunkedDiffs; 
+}
+
+// TODO: Simplify
+// TODO: Remove parts with '' at the end
+
+/**
+ * Merges the given list of diffs into a total diff,
+ * maintaining the inserts and deletes that happened
+ * in each diff.
+ */
+function mergeDiffs(diffs) {
+  if (diffs.length == 0) {
+    return diffs;
+  }
+
+  mergedDiff = JSON.parse(JSON.stringify(diffs[0]));
+  for (var i = 1; i < diffs.length; i++) {
+    var diff = JSON.parse(JSON.stringify(diffs[i]));
+
+    // Index into mergedDiff for what chunk we're currently on
+    var currentChunkInMerged = 0;
+
+    // Index within the current chunk
+    var indexInCurrentChunkInMerged = 0;
+
+    diff.forEach(function(part) {
+
+      if (part.added) {
+
+        var currentChunk = mergedDiff[currentChunkInMerged];
+
+        // Skip through the already removed chunks
+        // This preserves order if I remove something,
+        //   and then add something later in the same place
+        while (currentChunk && currentChunk.removed) {
+          currentChunkInMerged += 1;
+          currentChunk = mergedDiff[currentChunkInMerged];
+        }
+
+        if (!currentChunk) {
+          // The added part is at the very end of the file
+          mergedDiff.push(part);
+
+          // Our indexes are at the end of the last part
+          currentChunkInMerged = mergedDiff.length;
+          indexInCurrentChunkInMerged = part.value.length;
+
+        } else {
+          // Split up this chunk into previous and next
+          var prevChunk = JSON.parse(JSON.stringify(currentChunk));
+          prevChunk.value = prevChunk.value.substring(0, indexInCurrentChunkInMerged);
+          var nextChunk = JSON.parse(JSON.stringify(currentChunk));
+          nextChunk.value = nextChunk.value.substring(indexInCurrentChunkInMerged);
+
+          // Delete the current chunk and replace it with prev, part, and next
+          mergedDiff.splice(currentChunkInMerged, 1, prevChunk, part, nextChunk);
+
+          // Now, we start at the beginning of nextChunk
+          currentChunkInMerged += 2;
+          indexInCurrentChunkInMerged = 0;
+        }    
+        
+      } else if (part.removed) {
+
+        var currentChunk = mergedDiff[currentChunkInMerged];
+
+        // Skip through the already removed chunks
+        while (currentChunk && currentChunk.removed) {
+          currentChunkInMerged += 1;
+          currentChunk = mergedDiff[currentChunkInMerged];
+        }
+
+        if (indexInCurrentChunkInMerged + part.value.length < currentChunk.value.length) {
+          // The remove is within a single chunk
+          var prevChunk = JSON.parse(JSON.stringify(currentChunk));
```

Comment:
Just contract the object with the keys/values you know you want.

Reply:
Addressed w/ 2-list structure.